src/utils/outputs.ts

```diff
@@ -242,3 +242,99 @@ export const getSpeakingNames = (
   });
   return result;
 };
+
+export const getSpeakingNamesWithColors = (
+  speakingNames: SpeakingName[],
+  allColors: Record<string, string>,
+  colors: Record<string, HeisslufType[]>,
+  darkMode: boolean,
+  luminanceSteps: number[],
+  speakingNamesDefaultMapping: SpeakingName[]
+): any => {
+  let result: any = {};
+  let state;
+  const isStateName = (name: string) => {
+    return (
+      name.includes("enabled") ||
+      name.includes("hover") ||
+      name.includes("pressed")
+    );
+  };
+
+  const processState = (name: string) => {
+    const stateIndex = name.lastIndexOf("-");
+    state = name.slice(stateIndex + 1);
+    const nameWithoutState = name.slice(0, stateIndex);
+
+    const numOfState =
+      state === "enabled" ? "01" : state === "hover" ? "02" : "03";
+    const stateNumbered = `${numOfState}-${state}`;
+
+    return [nameWithoutState, numOfState, state, stateNumbered];
```

Comment:
Is there a reason why you return an array?

An object would be better {nameWithoutState, numOfState, state, stateNumbered}, I guess.